.github/workflows/release-on-milestone-closed.yml

```diff
@@ -0,0 +1,55 @@
+name: "Automatic Releases"
+
+on:
+  milestone:
+    types:
+      - "closed"
+
+jobs:
+  release:
+    name: "GIT tag, release & create merge-up PR"
+    runs-on: "ubuntu-20.04"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+
+      - name: "Release"
+        uses: "laminas/automatic-releases@1.0.1"
+        with:
+          command-name: "laminas:automatic-releases:release"
+        env:
+          "GITHUB_TOKEN": ${{ secrets.GITHUB_TOKEN }}
+          "SIGNING_SECRET_KEY": ${{ secrets.SIGNING_SECRET_KEY }}
+          "GIT_AUTHOR_NAME": ${{ secrets.GIT_AUTHOR_NAME }}
+          "GIT_AUTHOR_EMAIL": ${{ secrets.GIT_AUTHOR_EMAIL }}
+
+      - name: "Create Merge-Up Pull Request"
+        uses: "laminas/automatic-releases@1.0.1"
+        with:
+          command-name: "laminas:automatic-releases:create-merge-up-pull-request"
+        env:
+          "GITHUB_TOKEN": ${{ secrets.GITHUB_TOKEN }}
+          "SIGNING_SECRET_KEY": ${{ secrets.SIGNING_SECRET_KEY }}
+          "GIT_AUTHOR_NAME": ${{ secrets.GIT_AUTHOR_NAME }}
+          "GIT_AUTHOR_EMAIL": ${{ secrets.GIT_AUTHOR_EMAIL }}
+
+      - name: "Create and/or Switch to new Release Branch"
+        uses: "laminas/automatic-releases@v1"
+        with:
+          command-name: "laminas:automatic-releases:switch-default-branch-to-next-minor"
+        env:
+          "GITHUB_TOKEN": ${{ secrets.ORGANIZATION_ADMIN_TOKEN }}
+          "SIGNING_SECRET_KEY": ${{ secrets.SIGNING_SECRET_KEY }}
+          "GIT_AUTHOR_NAME": ${{ secrets.GIT_AUTHOR_NAME }}
+          "GIT_AUTHOR_EMAIL": ${{ secrets.GIT_AUTHOR_EMAIL }}
+
+      - name: "Create new milestones"
```

Comment:
You can remove this step if you don't want the milestone creation.

Reply:
What if I want it most of the time, but don't want it if I no longer maintain the branch?

Reply:
Close it manually :) If a milestone is empty when you close it, it does not create a release.

Now that I think of it, this might be an interesting feature: Set a LTS rule or something and autoclose a branch (if it's empty) after x days.

Reply:
>  If a milestone is empty when you close it, it does not create a release.

If it also does not create a milestone this should be OK, what do you think @morozov ? I kind of wish there was a way to mark a milestone as final, or a branch as end-of-life.

Reply:
Creating milestones manually is fine with me. In can be done right before assigning the first issue to it.

As for the final milestones, our experience with 2.10.3 shows that things don't always go as we plan in this regard, so as a maintainer of the release workflow, I wouldn't bother much about automating the EOL case.

Reply:
> Creating milestones manually is fine with me.

Did you mean to write "Closing"? Or do you want me to release that step entirely?

Reply:
I meant “creating”. I'm fine if this step is entirely removed unless it's a dependency for some other steps.

Reply:
Done 👍